### PR TITLE
[FLINK-27118][yarn] TM ignores localhost BIND_HOST

### PR DESCRIPTION
**Pull request by zentol**
Modifies the yarn taskexecutor runner to drop BIND_HOST configurations set to localhost, as done by the default distribution. 

Localhost doesn't work out-of-the-box, and the TM still using it wasn't intentional.

### CODE REVIEW

**Comment by zoltar9264** [comment]
Hi @zentol , I meet this problem on 1.15.0-rc0 , and I just set _'taskmanager.bind-host'_ to 0.0.0.0 before seeing this pr. 
Given the background I think this pr is handled appropriately, but I still recommend adding some clarification to flink-conf.yaml file to let users know that _'taskmanager.bind-host'_ doesn't take effect with yarn.

**Comment by gaoyunhaii** [comment]
@flinkbot run azure

**Comment by gaoyunhaii** [comment]
Thanks @zentol for the PR! We have a manual test and it should be able to solve the connection problem, but there is one issue  that in the TaskManager's log, it would print all the configurations, but it happens before the value is changed, thus it would still show `localhost` and might cause some confusion. Perhaps we could still change it before printing the configuration or in the JM side?

**Comment by zentol** [comment]
> Perhaps we could still change it before printing the configuration

This isn't possible since the printing is done when the config file is loaded, so any modifications must inherently occur afterwards.

This also occurs on the JM side when it loads the configuration, with no real way to solve it.

How about we just log an info message that we remove/change the bind-host setting?

**Comment by zentol** [comment]
I've tried another approach where we remove the options on the client instead. This should work for both JM/TM processes and prevent any settings that aren't actually applied from being logged.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java [hunk 1]
```diff
@@ -103,6 +103,11 @@ static void setupAndModifyConfiguration(
         BootstrapTools.updateTmpDirectoriesInConfiguration(configuration, localDirs);
 
         setupConfigurationFromVariables(configuration, currDir, variables);
+
+        configuration
```
Comment (gaoyunhaii) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java:107]: This should be unnecessary if we have already removed the configuration in the client side~?

Comment (gaoyunhaii) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java:107]: This should be unnecessary if we have already removed the configuration in the client side~?

**Review by gaoyunhaii — COMMENTED** [review]
Very thanks @zentol for the fix! I have a double manual test and it should work (both we also remove in the TM side or not).

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java (hunk 1 above)
Comment (zentol) [reply flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java]: yes

Comment (zentol) [reply flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskExecutorRunner.java]: es

**Review by gaoyunhaii — APPROVED** [review]
LGTM! Thanks @zentol for the fix and @zoltar9264 for the review!

**Pull request closed without merging**